src/herder/HerderImpl.cpp

```diff
@@ -1213,7 +1214,10 @@ HerderImpl::removeReceivedTxs(std::vector<TransactionFramePtr> const& dropTxs)
 void
 HerderImpl::recvSCPQuorumSet(Hash const& hash, const SCPQuorumSet& qset)
 {
-    mPendingEnvelopes.recvSCPQuorumSet(hash, qset);
+    if (isQuorumSetSane(qset, false))
+    {
+        mPendingEnvelopes.recvSCPQuorumSet(hash, qset);
+    }
 }
```

Comment:
The issue here is that now this request will time out and we will now ask the next peer.  If someone gets us to fetch a qset that we don't think is sane we will just keep asking for it. We should stop seeking it once we realize it isn't sane.